### PR TITLE
Gcloud integration

### PR DESCRIPTION
**Pull request by Mathieu4141**
Most of the added files here are to make the code compatible for gcloud, but there are a few things of interest for you @rezazzr, I'll explain those parts in comments

### CODE REVIEW

File: src/tricicl/loggers/tb.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+from matplotlib.figure import Figure
+from PIL.Image import Image
+from torch import Tensor
+from torch.utils.tensorboard import SummaryWriter
+from torchvision.transforms.functional import to_tensor
+
+from tricicl.utils.path import safe_path
+
+
+class TensorboardLogger(StrategyLogger):
```
Comment (Mathieu4141) [inline src/tricicl/loggers/tb.py:12]: This logger should work as well for AWS S3, by simply giving it the s3 path, but I never tested Tensorboard on it

File: src/tricicl/metrics/normalized_accuracy.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+from avalanche.evaluation.metrics import ExperienceAccuracy, StreamAccuracy
+
+"""
+Avalanche changed the behavior to the global training step 2w ago
+https://github.com/ContinualAI/avalanche/commit/89afc1cc620d88f09245920ac217a8be508e9e73#diff-6db76ef303657a86b96bc69cff62c8695bc4f046f683c43424ca8a2e608244f0
+which is not useful when plotting, getting the stream step is much more useful.
+"""
```
Comment (Mathieu4141) [inline src/tricicl/metrics/normalized_accuracy.py:7]: I realized that part after freezing the version, so I made this patch instead of changing the version again

File: src/tricicl/storage/remote_storage.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    def from_str(rel_path: str) -> "SyncPath":
+        return SyncPath.from_local(PROJECT_DIR / rel_path)
+
+
+class RemoteStorageABC(ABC):
+    def download_dataset(self, name: str):
+        self.download_zipped_directory_if_missing(get_dataset_sync_path(name))
+
+    def upload_dataset(self, name: str):
+        self.upload_zipped_directory(get_dataset_sync_path(name))
```
Comment (Mathieu4141) [inline src/tricicl/storage/remote_storage.py:34]: This is convenient because from gcloud downloading from a bucket is way faster than from a url. I guess it should be similar in AWS, if you want to benefit from those method (plus aggregation from S3 storage), you simply need to inherit `RemoteStorageABC` and implement the 3 basic abstract methods

**Comment by Mathieu4141** [comment]
[Adapt to gcloud](https://trello.com/c/SeMTUxKA/12-adapt-to-gcloud)
